velox/common/memory/tests/MemoryPoolTest.cpp

```diff
@@ -228,125 +230,220 @@ TEST(MemoryPoolTest, UncapMemory) {
   // caps are supported again.
 }
 
+std::vector<std::shared_ptr<IMemoryManager>> createManagers(int64_t quota) {
+  std::vector<std::shared_ptr<IMemoryManager>> managers;
+  managers.emplace_back(
+      std::make_shared<MemoryManager<MemoryAllocator>>(quota));
+  managers.emplace_back(
+      std::make_shared<MemoryManager<MmapMemoryAllocator>>(quota));
+  return managers;
+}
+
+MachinePageCount calculatePageNeeded(
+    const MappedMemory* mappedMemory,
+    MachinePageCount pageNeeded) {
+  auto& sizeClasses = mappedMemory->sizeClasses();
+  if (pageNeeded > sizeClasses.back()) {
+    return pageNeeded;
+  }
+  for (auto& sizeClass : sizeClasses) {
+    if (sizeClass >= pageNeeded) {
+      return sizeClass;
+    }
+  }
+  VELOX_UNREACHABLE();
+}
+
+void testMmapMemoryAllocation(
+    const MmapAllocator* mmapAllocator,
+    MachinePageCount allocPages,
+    size_t allocCount) {
+  MemoryManager<MmapMemoryAllocator> manager(8 * GB);
+  const auto kPageSize = 4 * KB;
+
+  auto& root = manager.getRoot();
+  auto& child = root.addChild("elastic_quota");
+
+  std::vector<void*> allocations;
+  uint64_t totalPageAllocated = 0;
+  uint64_t totalPageMapped = 0;
+  const auto pageIncremental = calculatePageNeeded(mmapAllocator, allocPages);
+  const auto isSmallAlloc = allocPages <= mmapAllocator->sizeClasses().back();
```

Comment:
shouldn't this be `mmapAllocator->sizeClasses().front();`

Reply:
It should be .back() because sizeClasses is an increasing sorted list. If the allocation is larger than our largest size class we use big alloc logic (contiguous allocation)